include/EASTL/vector.h

```diff
@@ -890,7 +890,19 @@ namespace eastl
 			else if(n < (size_type)(mpEnd - mpBegin))
 				resize(n);
 
+		#if EASTL_MOVE_SEMANTICS_ENABLED
+			this_type     temp;
+			size_type     nNewSize = size();
+			pointer const pNewData = DoAllocate(nNewSize);
+
+			pointer pNewEnd = eastl::uninitialized_move_ptr(mpBegin, mpEnd, pNewData);
+
+			temp.mpBegin    = pNewData;
+			temp.mpEnd      = pNewEnd;
+			temp.mpCapacity = pNewEnd;
+		#else
 			this_type temp(*this);  // This is the simplest way to accomplish this, 
```

Comment:
Yes, in my first implementation, I just used the move constructor with eastl::move(). With that, I have few UnitTest failed, because the move was done on the buffer, not one each element, so the the allocation at the end stays with the exact same size!
While the set_capicity() ask to reduce the size of the buffer/container to fit the size, it was not correct.

I'm agree that this implementation is quite long, and perhaps there is other internal vector I could call to reduce the code size. If you have idea for that, tell me.